From cb80f6b5026d028d5c67517d1d4c3a6387224724 Mon Sep 17 00:00:00 2001
From: FrancescoFerraro <francesco.f@variscite.com>
Date: Thu, 30 Nov 2023 12:15:47 +0100
Subject: [PATCH] camera: fix max resolution to 720p

Signed-off-by: FrancescoFerraro <francesco.f@variscite.com>
---
 camera/CameraDeviceHWLImpl.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/camera/CameraDeviceHWLImpl.cpp b/camera/CameraDeviceHWLImpl.cpp
index 7f7664b..8fe729f 100644
--- a/camera/CameraDeviceHWLImpl.cpp
+++ b/camera/CameraDeviceHWLImpl.cpp
@@ -291,6 +291,9 @@ status_t CameraDeviceHwlImpl::initSensorStaticData()
     int frmfps = 0;
     struct v4l2_frmsizeenum cam_frmsize;
     struct v4l2_frmivalenum vid_frmval;
+    char value[PROPERTY_VALUE_MAX] = {0};
+
+    property_get("ro.build.product", value, "0");
     while (ret == 0) {
         memset(TmpStr, 0, 20);
         memset(&cam_frmsize, 0, sizeof(struct v4l2_frmsizeenum));
@@ -355,6 +358,9 @@ status_t CameraDeviceHwlImpl::initSensorStaticData()
             mPreviewResolutions[previewCnt++] = cam_frmsize.discrete.width;
             mPreviewResolutions[previewCnt++] = cam_frmsize.discrete.height;
         }
+
+        if(cam_frmsize.discrete.height == 720 && (strncasecmp("som_mx8mn", value, strlen(value)) == 0))
+            break;
     }  // end while
 
     mPreviewResolutionCount = previewCnt;
-- 
2.43.0

