From 4fb01aff2ec67af69603423adc5639c8145008b1 Mon Sep 17 00:00:00 2001
From: FrancescoFerraro <francesco.f@variscite.com>
Date: Thu, 30 Nov 2023 12:29:16 +0100
Subject: [PATCH] Fix Auto focus Frame Freeze

Signed-off-by: FrancescoFerraro <francesco.f@variscite.com>
---
 camera/CameraDeviceHWLImpl.cpp | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/camera/CameraDeviceHWLImpl.cpp b/camera/CameraDeviceHWLImpl.cpp
index 8fe729f..60ac6e5 100644
--- a/camera/CameraDeviceHWLImpl.cpp
+++ b/camera/CameraDeviceHWLImpl.cpp
@@ -57,10 +57,6 @@ std::unique_ptr<CameraDeviceHwl> CameraDeviceHwlImpl::Create(
         return nullptr;
     }
 
-    ctrl_fd = open(*mDevPath[0], O_RDWR);
-    if (ctrl_fd < 0) {
-	ALOGE("%s invalid fd handle", __func__);
-    }
     ALOGI("%s: Created CameraDeviceHwlImpl for camera %u", __func__, device->camera_id_);
 
     return std::unique_ptr<CameraDeviceHwl>(device);
@@ -100,6 +96,10 @@ CameraDeviceHwlImpl::CameraDeviceHwlImpl(
 
     memset(mPreviewResolutions, 0, sizeof(mPreviewResolutions));
     memset(mPictureResolutions, 0, sizeof(mPictureResolutions));
+    ctrl_fd = open(*mDevPath[0], O_RDWR);
+    if (ctrl_fd < 0) {
+        ALOGE("%s invalid fd handle", __func__);
+    }
     memset(&caps_supports, 0, sizeof(caps_supports));
     m_raw_v4l2_format = -1;
     m_color_arrange = -1;
@@ -107,6 +107,9 @@ CameraDeviceHwlImpl::CameraDeviceHwlImpl(
 
 CameraDeviceHwlImpl::~CameraDeviceHwlImpl()
 {
+    if (ctrl_fd > 0)
+       close(ctrl_fd);
+
     if(m_meta) {
         delete m_meta;
         m_meta = NULL;
-- 
2.43.0

